docs/topics/streamfield.md

```diff
@@ -519,6 +519,33 @@ my_page.body.append(('paragraph', RichText("<p>And they all lived happily ever a
 my_page.save()
 ```
 
+If a block extending a StructBlock is to be used inside of the StreamField's value, the value of this block needs to be set using the block's `.to_python` method.
+
+```python
+
+from wagtail import blocks
+
+class UrlWithTextBlock(blocks.StructBlock):
+   url = blocks.URLBlock()
+   text = blocks.TextBlock()
+
+# using this block inside the content
+
+block = UrlWithTextBlock()
+data = {
+    'url': 'https://github.com/wagtail/',
+    'text': 'A very interesting and useful repo'
+}
+value = block.to_python(data)
```

Comment:
Are you sure this step is necessary? I'd expect passing the `data` dict directly to work here:

    my_page.body.append(('url', data))

Reply:
You're absolutely correct, of course. 

My (shoddy) detective work scouring the source code just got me so far in that I got my code to work like suggested in the commit, so I'm glad for your correction. I expected there had to be a more elegant solution.

 I'll adjust the PR right away. Thank you, @gasman!